src/backend/distributed/commands/vacuum.c

```diff
@@ -299,6 +320,29 @@ DeparseVacuumStmtPrefix(int vacuumFlags)
 		appendStringInfoString(vacuumPrefix, "VERBOSE,");
 	}
 
+#if PG_VERSION_NUM >= 120000
```

Comment:
This is crashing. I think we should either (a) add tests for the new features or (b) error out saying that we don't support it. Not only for vacuum, but also for any new thing that we allow.
 

```SQL
VACUUM (SKIP_LOCKED true )users_table;
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
Time: 1611.956 ms (00:01.612)
```

Reply:
Ideally we should be testing new features even if we error out on them. I'm considering adding a test that specifically goes through pg12 features, that way it'll be straight forward for having the pg11/pg10 results which spew errors

Reply:
Looking closer, `ANALYZE` & `VERBOSE` are both applicable to `VACUUM` while also being handled later on in the code. So this assert is straight forward to hit with `VACUUM (ANALYZE)`

I've removed the assertion

Reply:
I don't think so, shouldn't we just need to add `VACOPT_SKIP_LOCKED ` to `unsupportedFlags `? We're already negating during the assignment? ```const int unsupportedFlags PG_USED_FOR_ASSERTS_ONLY = ~(...```

So, I'd rather add the assert back.

Reply:
No, because the idea behind `unsupportedFlags` is that any of the flags before then will be removed or we'll've raised an error. It turns out the only flag that shouldn't remain by time we get to the assert is VACOPT_VACUUM